src/lib/locationHelpers.ts

```diff
@@ -0,0 +1,37 @@
+import { isString } from "lodash"
+
+interface Location {
+  lat: number
+  lng: number
+  maxDistance?: number
+}
+
+const DEFAULT_MAX_DISTANCE_KM = 75
+
+export const getLocationArgs = async (
+  near: Location | undefined,
+  ip: string | undefined,
+  requestLocationLoader: any
+) => {
+  let location = near
+
+  if (ip) {
+    const {
+      body: { data: locationData },
+    } = await requestLocationLoader({ ip })
+
+    if (locationData.location) {
+      location = {
+        lat: locationData.location.latitude,
+        lng: locationData.location.longitude,
+      }
+    }
+  }
+
+  if (!location) return {}
+
+  return {
+    near: isString(near) ? near : `${location.lat},${location.lng}`,
+    max_distance: location.maxDistance || DEFAULT_MAX_DISTANCE_KM,
+  }
+}
```

Comment:
**thought:** I was expecting the business logic for the location to be inside Gravity. It should help us also potentially not make an extra API request

Reply:
Wouldn't this require us to make an external API call from Gravity to get an IP-based location? I guess Node is generally a better choice than Rails for making API calls within a request.